### DISCUSSION TITLE
globalFilter wiht fuzzy search

### DISCUSSION THREAD

**Discussion by marcok81** [TanStack/table#2196] [General]
Hi,

so first of all thanks for this great module. Now to my question:
**I would like to use a GlobalFilter to search my entire table with a fuzzy search.**

Ho can i pass my fuzzy search filtertype as a globalFilter type?

Maybe someone has an example.

Thanks a lot

**Comment by scottwernervt** [comment]
Here is a simple global text search example:

https://codesandbox.io/s/react-table-global-fuzzy-text-search-2y55j2?file=/src/App.js

```tsx
const globalFilter = useCallback(
  (rows: Row<RowData>[], ids: IdType<string>[], query: string) => {
    const searchTerm = String(query).toLowerCase();
    return rows.filter((row) => {
      const matches = ids.filter((id) => {
        const rowValue = row.values[id];
        return rowValue !== undefined
          ? String(rowValue).toLowerCase().includes(searchTerm)
          : false;
      });

      return matches.length > 0;
    });
  },
  [],
);
```

![Kapture 2022-05-06 at 09 50 39](https://user-images.githubusercontent.com/1048949/167147209-9dee84c2-a186-41a6-8eac-13126e05936e.gif)

Check out https://github.com/kentcdodds/match-sorter if you need more advanced fuzzy text searching.

**Reply by Muzi09** [reply]
Hy, how can i disable fuzzy search and get only exact data which matches the search query.
